### PR TITLE
Added docs for signing/verifying commits

### LINKED ISSUE #2249: New dev requirement: digitialy signed commits
### Summary of Issue

We should move to having all git commits digitally signed. It improves saftey of the codebase and prevents weird github attacks where someone fakes an ID to commit malicious code. 

- figure out how to make this work 
- document the process so outside users can also sign their commits and PRs
- put the new docs somewhere in the OM docs.

### PR DESCRIPTION
**Pull request by tadkollar**
### Summary

This documents how to configure GPG and GitHub to sign and verify commits. Once everyone on the dev team is set up for it, we can enable "Require signed commits" in the protection rule for the `master` branch. Note that PRs submitted before that point that are still open will need to have their unsigned commits overwritten with verified commits (covered in the doc).

### Related Issues

- Resolves #2249

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/docs/openmdao_book/other_useful_docs/developer_docs/signing_commits.ipynb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    "The process is [documented in detail by GitHub](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/about-commit-signature-verification). Here are the highlights:\n",
+    "\n",
+    "1. Install GPG if necessary\n",
+    "    - Source and binary releases are available at [gnupg.org](https://gnupg.org/download/)\n",
+    "    - Also available through many OS package managers. Examples:\n",
+    "        - MacOS: `brew install gpg`\n",
+    "        - RPM-based Linux distro: `sudo yum install gnupg2` or `sudo dnf install gnupg2`\n",
+    "        - DEB-based Linux distro: `sudo apt install gnupg`\n",
+    "        - Windows: `scoop bucket add nonportable`, then `scoop install gpg-np` ([Obtain Scoop](https://scoop.sh/))\n",
+    "2. [Generate a new GPG key](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/generating-a-new-gpg-key)\n",
```
[the remark this answers is not in the document]
Comment (tadkollar) [reply openmdao/docs/openmdao_book/other_useful_docs/developer_docs/signing_commits.ipynb:27]: That link is already in the title of the section:
`
2. [Generate a new GPG key](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/generating-a-new-gpg-key)
    - It's important to note that the email address used for the key must be identical to your GitHub commit email address
`

[the remark this answers is not in the document]
Comment (Kenneth-T-Moore) [reply openmdao/docs/openmdao_book/other_useful_docs/developer_docs/signing_commits.ipynb:27]: Wonder how I missed that...

**Review by Kenneth-T-Moore — APPROVED** [review]

**Review by naylor-b — APPROVED** [review]

File: openmdao/docs/openmdao_book/other_useful_docs/developer_docs/signing_commits.ipynb [hunk 2]
```diff
@@ -0,0 +1,100 @@
+{
```
Comment (swryan) [inline openmdao/docs/openmdao_book/other_useful_docs/developer_docs/signing_commits.ipynb:1]: <p>I had to <code>git config user.signingkey XXXXXXXX</code></p><p> </p><p>https://stackoverflow.com/questions/36810467/git-commit-signing-failed-secret-key-not-available</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMDAO/OpenMDAO/pull/2273/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (swryan) [reply openmdao/docs/openmdao_book/other_useful_docs/developer_docs/signing_commits.ipynb:1]: <p>fair enough.. I was just following your outline (and clicking through to the links) but missed that step</p>

**Review by swryan — APPROVED** [review]

**Comment by Kenneth-T-Moore** [comment]
Here is the command that the `git` docs didn't mention:

`git config user.signingkey s0e9fseyourkeyhere`

**Pull request merged**